src/storage/sftp/storage.c

```diff
@@ -831,31 +832,52 @@ storageSftpNew(const String *const path, const String *const host, const unsigne
             THROW_FMT(ServiceError, "libssh2 handshake failed");
 
         int hashType = LIBSSH2_HOSTKEY_HASH_SHA1;
+        size_t hashSize = 0;
 
-        switch (param.hostkeyHash)
+        switch (param.hostkeyHashType)
         {
             case hashTypeMd5:
                 hashType = LIBSSH2_HOSTKEY_HASH_MD5;
+                hashSize = 16;
                 break;
 
             case hashTypeSha1:
                 hashType = LIBSSH2_HOSTKEY_HASH_SHA1;
+                hashSize = 20;
                 break;
 
 #ifdef LIBSSH2_HOSTKEY_HASH_SHA256
             case hashTypeSha256:
                 hashType = LIBSSH2_HOSTKEY_HASH_SHA256;
+                hashSize = 32;
                 break;
 #endif // LIBSSH2_HOSTKEY_HASH_SHA256
 
             default:
-                THROW_FMT(ServiceError, "requested ssh2 hostkey hash type (%s) not available", strZ(strIdToStr(param.hostkeyHash)));
+                THROW_FMT(
+                    ServiceError, "requested ssh2 hostkey hash type (%s) not available", strZ(strIdToStr(param.hostkeyHashType)));
                 break;
         }
 
-        if (libssh2_hostkey_hash(driver->session, hashType) == NULL)
+        const char *binaryFingerprint = libssh2_hostkey_hash(driver->session, hashType);
+
+        if (binaryFingerprint == NULL)
             THROW_FMT(ServiceError, "libssh2 hostkey hash failed: libssh2 errno [%d]", libssh2_session_last_errno(driver->session));
 
+        // Compare fingerprint if provided
+        if (param.hostFingerprint != NULL)
+        {
+            char fingerprint[256];
+
+            encodeToStr(encodingHex, (unsigned char *)binaryFingerprint, hashSize, fingerprint);
+
+            if (strcmp(fingerprint, strZ(param.hostFingerprint)) != 0)
+            {
+                THROW_FMT(
+                    ServiceError, "host [%s] and provided fingerprint [%s] do not match", fingerprint, strZ(param.hostFingerprint));
```

Comment:
I'd be inclined to say 'configured' here and include the option name, to make it clear what we're talking about.  eg:

host [%s] and configured fingerprint (repo-sftp-host-fingerprint) [%s] do not match

Reply:
Addressed in d080b7f9071478ed0730f5ee15efe0b4cfba13b4